mne_bids/dig.py

```diff
@@ -281,24 +280,24 @@ def _write_coordsystem_json(*, raw, unit, hpi_coord_system,
         Defaults to False.
 
     """
-    dig = raw.info['dig']
-    if dig is None:
-        dig = []
-
-    coord_frame = set([dig[ii]['coord_frame'] for ii in range(len(dig))])
-    if len(coord_frame) > 1:  # noqa E501
-        raise ValueError('All HPI, electrodes, and fiducials must be in the '
-                         'same coordinate frame. Found: "{}"'
-                         .format(coord_frame))
+    if raw.get_montage() is None or raw.get_montage().get_positions() is None:
+        dig = list()
+        coords = dict()
+    else:
+        montage = raw.get_montage()
+        pos = montage.get_positions()
+        dig = list() if montage.dig is None else montage.dig
+        coords = dict(
+            NAS=list() if pos['nasion'] is None else pos['nasion'].tolist(),
+            LPA=list() if pos['lpa'] is None else pos['lpa'].tolist(),
+            RPA=list() if pos['rpa'] is None else pos['rpa'].tolist())
 
     # get the coordinate frame description
     sensor_coord_system_descr = (BIDS_COORD_FRAME_DESCRIPTIONS
                                  .get(sensor_coord_system.lower(), "n/a"))
 
-    coords = _extract_landmarks(dig)
```

Comment:
What's wrong with `_extract_landmarks`?

Reply:
It's redundant with `raw.get_montage()` which checks the same things. Seems wise to use the checks there rather than reimplementing them.